pyxem/signals/diffraction_vectors.py

```diff
@@ -260,21 +266,110 @@ def from_peaks(
         vectors.column_names = column_names
         return vectors
 
+    def subpixel_refine(
+        self,
+        signal,
+        method="center-of-mass",
+        disk_r=None,
+        upsample_factor=2,
+        square_size=10,
+        **kwargs,
+    ):
+        """
+        Refine the positions of the diffraction vectors using subpixel
+        interpolation.
+
+        Parameters
+        ----------
+        signal : Signal
+            The signal which will be used to refine the diffraction vectors.
+        method : str
+            The method used to refine the diffraction vectors. Currently
+            supported methods are 'cross-correlation' and 'phase-correlation',
+            and "center-of-mass".
+
+
+        Returns
+        -------
+        refined_vectors : DiffractionVectors
+            The refined vectors.
+        """
+        method_dict = {
+            "cross-correlation": _conventional_xc_map,
+            "center-of-mass": _center_of_mass_map,
+        }
+        if method not in method_dict:
+            raise ValueError(
+                f"The method parameter must be one of {list(method_dict.keys())}"
+            )
+        if method == "center-of-mass":
+            if disk_r is not None or upsample_factor is not None:
```

Comment:
At present this emits on the default call to the method. I'm not sure we should be doing this sort of checking either way.

Reply:
You're right it should be fine without!